drv/stm32h7-update-server/src/main.rs

```diff
@@ -387,10 +387,10 @@ impl idl::InOrderUpdateImpl for ServerImpl<'_> {
                 .read_range(0..len, flash_bytes)
                 .map_err(|_| RequestError::Fail(ClientError::WentAway))?;
 
-            // If there is a write less than the block size zero out the
-            // trailing bytes
+            // If there is a write less than the block size fill out the
+            // trailing bytes with 0xff.
             if len < BLOCK_SIZE_BYTES {
-                flash_bytes[len..].fill(0);
+                flash_bytes[len..].fill(0xff);
```

Comment:
Nit: could we initialize it to `0xFF` instead, then skip this step entirely?

Reply:
Looking at the BorrowedBuffer implementation, It would be more work but less branching/code paths. It depends on what you want to optimize for.
In this case, I'm going for smallest, easiest to approve change.

Reply:
I'm referring to `let mut flash_page = ...` above, which initializes the nested arrays to all zeros.